pkg/kubectl/describe.go

```diff
@@ -94,12 +98,16 @@ func DescribableResources() []string {
 
 // Describer returns the default describe functions for each of the standard
 // Kubernetes types.
-func DescriberFor(kind string, c *client.Client) (Describer, bool) {
-	f, ok := describerMap(c)[kind]
-	if ok {
-		return f, true
+func DescriberFor(kind string, c client.VersionInterface) (Describer, bool) {
```

Comment:
DescriberFor should probably just take both clients.  This method is
already flawed (assuming that there is a mapping between the Kind provided
and a client).

On Fri, Aug 7, 2015 at 2:06 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/kubectl/describe.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12405#discussion_r36544649>
> :
>
> > @@ -94,12 +98,16 @@ func DescribableResources() []string {
> >
> >  // Describer returns the default describe functions for each of the standard
> >  // Kubernetes types.
> > -func DescriberFor(kind string, c *client.Client) (Describer, bool) {
> > -	f, ok := describerMap(c)[kind]
> > -	if ok {
> > -		return f, true
> > +func DescriberFor(kind string, c client.VersionInterface) (Describer, bool) {
>
> I understand you use VersionInterface because it's a common interface
> implemented by both Client and ExpClient, but that's not the supposed use
> of VersionInterface. Can we define an interface similar to IsAnAPIObject?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12405/files#r36544649>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift